### PR TITLE
Execution model fields

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Updates ExecutionClosure with optional AbortMetadata and ExecutionMetadata with the complete set of fields

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
User requested feature

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2881

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/models/execution.py [hunk 1]
```diff
@@ -13,52 +20,130 @@
...
+
+    @property
+    def execution_cluster(self) -> str:
+        return self._execution_cluster
+
+    def to_flyte_idl(self) -> flyteidl.admin.execution_pb2.SystemMetadata:
+        return _execution_pb2.SystemMetadata(execution_cluster=self.execution_cluster)
+
+    @classmethod
+    def from_flyte_idl(cls, pb2_object) -> SystemMetadata:
```
Comment (wild-endeavor) [inline flytekit/models/execution.py:35]:
```suggestion
    def from_flyte_idl(cls, pb2_object: flyteidl.admin.execution_pb2.SystemMetadata) -> SystemMetadata:
```

File: flytekit/models/execution.py [hunk 2]
```diff
@@ -13,52 +20,130 @@
...
+        """
+        :param flyteidl.admin.execution_pb2.SystemMetadata pb2_object:
```
Comment (wild-endeavor) [inline flytekit/models/execution.py:37]: kill

File: flytekit/models/execution.py [hunk 3]
```diff
@@ -13,52 +20,130 @@
...
         SCHEDULED = 1
         SYSTEM = 2
 
-    def __init__(self, mode, principal, nesting):
+    def __init__(
+        self,
+        mode: int,
+        principal: str,
+        nesting: int,
+        scheduled_at: datetime.datetime = None,
```
Comment (wild-endeavor) [inline flytekit/models/execution.py:56]:
```suggestion
        scheduled_at: typing.Optional[datetime.datetime] = None,
```
and the other three?

**Review by wild-endeavor — APPROVED** [review]
nits, if you want to fix them

File: flytekit/models/execution.py (hunk 2 above)
Comment (katrogan) [reply flytekit/models/execution.py:37]: copy pasta, thanks

File: flytekit/models/execution.py (hunk 3 above)
Comment (katrogan) [reply flytekit/models/execution.py]: done

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**